### ISSUE TITLE
Magisk modules clean installation issue

### ISSUE THREAD

**Issue by yongcrypt** [topjohnwu/Magisk#3314]
There is a case when a new ROM is flashed (using TWRP) with a clean installation method (wipe cache/dalvik-cache, data, etc.) and Magisk right after that. In this case, NVBASE will be set to /data (or /cache/data_adb) according to the data and DE storage writable checks here:
https://github.com/topjohnwu/Magisk/blob/master/scripts/util_functions.sh#L516-L528
and MAGISKBIN uses that NVBASE variable:
https://github.com/topjohnwu/Magisk/blob/master/scripts/util_functions.sh#L57-L58
In all other cases MAGISKBIN will be /data/adb/magisk, but (it looks like) in the first case it also gets updated from /data/magisk to /data/adb/magisk due to a forced reboot during first boot. I don't known how and where this reboot happens to be honest, but well...

However, the module_installer.sh script does not cover this clean installation case and always expects all Magisk stuff to be in /data/adb/magisk: https://github.com/topjohnwu/Magisk/blob/master/scripts/module_installer.sh#L28-L29 , so the module's installation fails because of this.

Simply put, you can do a clean installation of the ROM and flash (at the same time) Magisk after that, but you can't also flash required modules (e.g. for permissive selinux) due to hardcoded /data/adb/magisk in module_installer.sh. Again, for clarification, this issue happens only if /data was wiped before the ROM and Magisk installation.

I really hope I was able to explain this correctly, but there still may be inaccuracies somewhere. Please clarify if something is unclear.

**Comment by topjohnwu** [comment]
It is not possible to install Magisk modules cleanly exactly because of the reason of `/data/adb` access. This is a limitation of FBE
